### PR TITLE
Added query validation and instrumentation to Nadel

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

File: src/main/java/graphql/nadel/Nadel.java [hunk 1]
```diff
@@ -1,69 +1,91 @@
...
+        this.services = serviceList;
+
+        List<DefinitionRegistry> registries = serviceList.stream()
                 .map(Service::getDefinitionRegistry)
-                .collect(Collectors.toList());
+                .collect(toList());
+        OverallSchemaGenerator overallSchemaGenerator = new OverallSchemaGenerator();
         this.overallSchema = overallSchemaGenerator.buildOverallSchema(registries);
-        this.execution = new Execution(services, overallSchema);
-
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/Nadel.java]: removing the init method and making things `final`

File: src/main/java/graphql/nadel/NadelExecutionInput.java [hunk 2]
```diff
@@ -16,10 +17,10 @@
     private final Map<String, Object> variables;
 
     private NadelExecutionInput(String query, String operationName, Object context, Map<String, Object> variables) {
-        this.query = query;
+        this.query = requireNonNull(query);
         this.operationName = operationName;
         this.context = context;
-        this.variables = variables;
+        this.variables = requireNonNull(variables);
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/NadelExecutionInput.java]: andi raised a bug on this

File: src/main/java/graphql/nadel/Operation.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
     private final String displayName;
     private final OperationDefinition.Operation astOperation;
 
-    private Operation(String name, String displayName, OperationDefinition.Operation astOperation) {
+    Operation(String name, String displayName, OperationDefinition.Operation astOperation) {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/Operation.java]: idea told me to - squiggle line

File: src/test/groovy/graphql/nadel/OverallSchemaGeneratorTest.groovy [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package graphql.nadel
 
-
+import graphql.nadel.testutils.TestUtil
```
Comment (bbakerman) [inline src/test/groovy/graphql/nadel/OverallSchemaGeneratorTest.groovy]: moved this test code

File: src/main/java/graphql/nadel/NadelExecutionInput.java [hunk 5]
```diff
@@ -16,10 +17,10 @@
     private final Map<String, Object> variables;
 
     private NadelExecutionInput(String query, String operationName, Object context, Map<String, Object> variables) {
-        this.query = query;
+        this.query = requireNonNull(query);
```
Comment (andimarek) [inline src/main/java/graphql/nadel/NadelExecutionInput.java]: we must decide if we wanna got with the graphq-java assertions or with java ones.

I used the graphql-java ones so far

Comment (bbakerman) [reply src/main/java/graphql/nadel/NadelExecutionInput.java]: I saw the Java ones and copied that.  

For null asserts I would use Java ones (post 1.8) but Java 8 does not have the other types so I think a mix of Java NPE ones plus other ones is ok

**Comment by bbakerman** [comment]
Fair call on the E2E test size

**Pull request merged**